### DISCUSSION TITLE
Updating subscription variables

### DISCUSSION THREAD

**Discussion by DoisKoh** [enisdenjo/graphql-ws#541] [Q&A] [answered]
Is it possible to update an existing subscription's variables? I'm getting the issue where if my WebSocket connection terminates, and graphql-ws retries the connection, I need to continue the subscription with new variables (the variable is basically a timestamp for a cursor to determine what data gets pulled).

If it's relevant, I'm using urql on top of graphql-ws...

**Comment by DoisKoh** [comment]
Is the only way for me to deal with this to drop the subscription and create a new one with updated variables?

**Reply by enisdenjo** [reply, marked as answer]
Actually yes. An alternative would be to do some JS trickery with [getters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get).

**Reply by DoisKoh** [reply]
Welp, anyway if it's helpful to anyone out there, I ended up dropping the subscriptions and recreating them with new variables on connection close & connection connected events from `graphql-ws`.

I didn't consider getters but I wish I had... sounds like it would be a better idea than all the boilerplate I've had to add for each subscription.
